websites/R/RTLplay/util.ts

```diff
@@ -0,0 +1,325 @@
+			break;
+		}
+		case "de-DE": {
+			strings.deferred = "Zeitversetzt";
+			strings.movie = "Film";
+			strings.tvshow = "TV-Serie";
+			strings.privacy = "Private Mode";
+			strings.watchingLiveMusic = "Schaut ein Musikvideo live";
+			break;
+		}
+		default: {
+			strings.deferred = "Deferred";
+			strings.movie = "Movie";
+			strings.tvshow = "TV Serie";
+			strings.privacy = "Privacy Mode";
+			strings.watchingLiveMusic = "Watching a live music video";
+		}
+	}
+	return strings;
+}
+
+export const enum LargeAssets {
+	Logo = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/logo.png",
+	Animated = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/0.gif",
+	Deferred = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/1.gif",
+	LiveAnimated = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/2.gif",
+	Vinyle = "https://i.imgur.com/6qvsVLa.png",
+	VinyleAnimated = "https://i.imgur.com/8nd4UdO.gif",
+	Binoculars = "https://imgur.com/aF3TWVK.png",
+	Privacy = "https://imgur.com/nokHvhE.png",
+	AdEn = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/4.png",
+	AdFr = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/5.png",
+	RTLPlay = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/6.png",
+	RTLTVi = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/7.png",
+	RTLClub = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/8.png",
+	RTLPlug = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/9.png",
+	RTLDistrict = "https://i.imgur.com/gZdD4LA.png",
+	BelRTL = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/10.png",
+	Contact = "https://cdn.rcd.gg/PreMiD/websites/R/RTLplay/assets/11.png",
+}
+
+export function getLocalizedAssets(
+	lang: string,
+	assetName: string
+): LargeAssets {
+	switch (assetName) {
+		case "Ad":
+			switch (lang) {
+				case "fr-FR":
+					return LargeAssets.AdFr;
+				default:
+					return LargeAssets.AdEn;
+			}
+		default:
+			return LargeAssets.Binoculars; // Default fallback
+	}
+}
+
+// Mainly used to truncate largeImageKeyText because the limit is 128 characters
+export function limitText(input: string, maxLength = 128): string {
+	const ellipsis = " ...";
+
+	// If input is within limit, return it as is
+	if (input.length <= maxLength) return input;
+
+	// Truncate to 125 characters (leaving room for ellipsis)
+	let truncated = input.slice(0, maxLength - ellipsis.length);
+
+	// If the truncated text ends mid-word, remove the partial word
+	if (truncated.lastIndexOf(" ") !== -1)
+		truncated = truncated.slice(0, truncated.lastIndexOf(" "));
+
+	return truncated + ellipsis;
+}
+
+export function exist(selector: string): boolean {
+	return document.querySelector(selector) !== null;
+}
+
+// Copy of the function in Youtube utils
+let cachedTime = 0;
+export function adjustTimeError(time: number, acceptableError: number): number {
+	if (Math.abs(time - cachedTime) > acceptableError) cachedTime = time;
+	return cachedTime;
+}
```

Comment:
this shouldn't be needed anymore.

Reply:
Hi @Bas950 ! 
I wanted to add the features keeping in mind that it works correctly on older versions too. I noticed that with Version: 2.7.0 (ba99913), when a media is played, the extension spammed headless_sessions in the service worker because the start timestamp was updated too fast, resulting in a lot of headless_sessions failing in rate limited. This meant that the change to other pages was much less responsive or was blocking or taking a long time because I guess there's probably a queue set up for status discord changes. So I looked at how other presences handled timestamps, found this function and it solved the problem. The headless_sessions stabilize after a while and only send when necessary, and the page change is more responsive.

I also tested with the latest recoded build Version: 2.7.0 (e9ff721) if this problem existed and it doesn't. So I agree with you and I came to the conclusion that this function was temporary and could be removed in the future. During development I went back and forth between the 2 versions to see if everything worked as I imagined it would in both.

With adjust function
![2 7 0_adjust](https://github.com/user-attachments/assets/69a0b0c9-dcdc-4778-93ad-cd4428a56802)
Without
![2 7 0_noadjust](https://github.com/user-attachments/assets/2c07ce4f-89cb-4672-a63c-386ae17079aa)

Reply:
@Timeraa might be able to confirm, but I don't think we will release these presences to the current bundled extension version. So it will be released with the new alpha/beta version.

Reply:
I'm not quite sure, I hope to finish 2.7.0 this year but no promises there but the spamming issue still somewhat happens with that function on 2.6.x because of a broken if check